code/modules/vehicle/vehicle.dm

```diff
@@ -80,48 +81,38 @@
 		if(A.density)
 			if(A != src && A != M)
 				return
-	M.loc = get_turf(src)
+	M.forceMove(get_turf(src))
 	..()
 	handle_vehicle_offsets()
 
-
 /obj/vehicle/bullet_act(obj/item/projectile/Proj)
-	if(buckled_mob)
-		buckled_mob.bullet_act(Proj)
+	if(has_buckled_mobs())
+		for(var/m in buckled_mobs)
+			var/mob/living/buckled_mob = m
+			buckled_mob.bullet_act(Proj)
 
 //MOVEMENT
 /obj/vehicle/relaymove(mob/user, direction)
 	if(user.incapacitated())
-		unbuckle_mob()
+		unbuckle_mob(user)
+		return
 
+	if(world.time < next_vehicle_move)
+		return
+	next_vehicle_move = world.time + vehicle_move_delay
 	if(keycheck(user))
-		if(!Process_Spacemove(direction) || world.time < next_vehicle_move || !isturf(loc))			return
-		next_vehicle_move = world.time + vehicle_move_delay
-
+		if(!Process_Spacemove(direction) || !isturf(loc))
+			return
 		step(src, direction)
 
-		if(buckled_mob)
```

Comment:
this ugly hack may have to stay; the reason we have it is because our slipping is handled completely differently.

Reply:
Ah right, hence that component PR.